### PR TITLE
Feat/support vpc network

### PR DESCRIPTION
**Pull request by dactoankmapydev**
- Add API get, create, update, delete network interface in VPC network
- Add API list network interfaces 
- Add test case

### CODE REVIEW

File: vpc.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ type VPCService interface {
 	Update(ctx context.Context, vpcID string, uvpl *UpdateVPCPayload) (*VPC, error)
 	Create(ctx context.Context, cvpl *CreateVPCPayload) (*VPC, error)
 	Delete(ctx context.Context, vpcID string) error
+	CreateVPCNetworkInterface(ctx context.Context, networkID string, cnip *CreateNetworkInterfacePayload) (*NetworkInterface, error)
+	UpdateVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string, unip *UpdateNetworkInterfacePayload) (*NetworkInterface, error)
+	DeleteVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string) error
+	GetVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string) (*NetworkInterface, error)
```
Comment (greatbn) [inline vpc.go:31]: we only need netInterfaceID when get detail of a network interface

File: vpc.go [hunk 2]
```diff
@@ -25,6 +25,11 @@ type VPCService interface {
 	Update(ctx context.Context, vpcID string, uvpl *UpdateVPCPayload) (*VPC, error)
 	Create(ctx context.Context, cvpl *CreateVPCPayload) (*VPC, error)
 	Delete(ctx context.Context, vpcID string) error
+	CreateVPCNetworkInterface(ctx context.Context, networkID string, cnip *CreateNetworkInterfacePayload) (*NetworkInterface, error)
+	UpdateVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string, unip *UpdateNetworkInterfacePayload) (*NetworkInterface, error)
```
Comment (greatbn) [inline vpc.go:29]: ditto

File: vpc.go [hunk 3]
```diff
@@ -25,6 +25,11 @@ type VPCService interface {
...
+	DeleteVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string) error
+	GetVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string) (*NetworkInterface, error)
+	ListNetworkInterface(ctx context.Context) ([]*NetworkInterface, error)
```
Comment (greatbn) [inline vpc.go:32]: with list, I think you can pass search options for this. such as list network interface by VPC Network ID or list all network interface

File: vpc.go [hunk 4]
```diff
@@ -180,3 +238,83 @@ func (v vpcService) Delete(ctx context.Context, vpcID string) error {
...
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (v vpcService) GetVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string) (*NetworkInterface, error) {
+	req, err := v.client.NewRequest(ctx, http.MethodGet, serverServiceName, v.resourceVPCNetworkInterfacePath(networkID, netInterfaceID), nil)
```
Comment (greatbn) [inline vpc.go:260]: Do we need networkID here ?

File: vpc.go [hunk 5]
```diff
@@ -180,3 +238,83 @@ func (v vpcService) Delete(ctx context.Context, vpcID string) error {
...
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (v vpcService) UpdateVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string, unip *UpdateNetworkInterfacePayload) (*NetworkInterface, error) {
+	req, err := v.client.NewRequest(ctx, http.MethodPut, serverServiceName, v.resourceVPCNetworkInterfacePath(networkID, netInterfaceID), unip)
```
Comment (greatbn) [inline vpc.go:277]: ditto

File: vpc.go [hunk 6]
```diff
@@ -180,3 +238,83 @@ func (v vpcService) Delete(ctx context.Context, vpcID string) error {
...
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (v vpcService) DeleteVPCNetworkInterface(ctx context.Context, networkID string, netInterfaceID string) error {
```
Comment (greatbn) [inline vpc.go:293]: ditto.

**Comment by dactoankmapydev** [comment]
switch to new branch feat/support-network-interface

**Pull request closed without merging**